src/Datadog.Trace.ClrProfiler.Managed/Emit/ObjectExtensions.cs

```diff
@@ -231,31 +245,33 @@ private static PropertyFetcherCacheKey GetKey<TResult>(string name, Type type)
                 return null;
             }
 
-            var dynamicMethod = Emit<Func<object, TResult>>.NewDynamicMethod($"{containerType.FullName}.{fieldName}");
-            dynamicMethod.LoadArgument(0);
+            DynamicMethod dynamicMethod = new DynamicMethod($"{containerType.FullName}.{fieldName}", typeof(TResult), new Type[] { typeof(object) }, ObjectExtensions.Module, skipVisibility: true);
+            ILGenerator il = dynamicMethod.GetILGenerator();
+
+            il.Emit(OpCodes.Ldarg_0);
 
             if (containerType.IsValueType)
             {
-                dynamicMethod.UnboxAny(containerType);
+                il.Emit(OpCodes.Unbox, containerType);
```

Comment:
the conversion should be `Unbox_Any`

Reply:
Thanks for spotting this! Fixed in 0ce85d3fe2582b68df47c0e74f64d295476ccf14